source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -48,8 +44,21 @@ method of any cursor to iterate through results in a functional style:
    :start-after: begin forEachRemainingIteration
    :end-before: end forEachRemainingIteration
 
-Manual Iteration
-~~~~~~~~~~~~~~~~
+.. note::
+
+   The ``find()`` method implements the Iterable interface. We often see the
+   ``forEach()`` method used to iterate through results:
+
+   .. code-block:: java
+
+      collection.find().forEach(doc -> System.out.println(doc.toJson()));
+
+   Using this method is not preferable because you can't ensure that
```

Comment:
This actually isn't true.  The driver's Iterable implementation of forEach ensures the cursor is closed. We might, however, might want to warn against use of the enhanced for loop syntax:

    for (Document cur : collection.find()) {
         ..
    }

If there is an exception thrown before iteration completes, the cursor will _not_ be closed.

If you want to see the implementation, it's [here](https://github.com/mongodb/mongo-java-driver/blob/master/driver-sync/src/main/com/mongodb/client/internal/MongoIterableImpl.java#L120):